tests/executables_src/testScalarRegisterAccessor.cpp

```diff
@@ -135,6 +162,15 @@ BOOST_AUTO_TEST_CASE(testIntRegisterAccessor) {
   accessor.write();
   BOOST_CHECK(dummy == 119);
 
+  // test readAndGet
+  accessor = 470;
+  accessor.write();
+  BOOST_CHECK(accessor.readAndGet() == 470);
```

Comment:
Hm, somehow a review comment has not been saved.

This line is only testing half of the story: It is only testing that the content of the buffer is returned. But it does not test the reading is happening because the buffer was 470 already before.

The correct implementation would be
```
dummy = 470;
BOOST_CHECK(accessor.readAndGet() == 470);
```